src/Models/AiCreditTopup.php

```
@@ -0,0 +1,224 @@
+     */
+    public static function saveAutoTopupSettings(Database $db, array $data): void
+    {
+        $orgId = self::positiveId($data['org_id'] ?? null, 'org_id');
+        $userId = self::positiveId($data['updated_by_user_id'] ?? null, 'updated_by_user_id');
+        $enabled = !empty($data['enabled']) ? 1 : 0;
+        $thresholdCredits = self::boundedNonNegativeInt($data['threshold_credits'] ?? 0, 1000000, 'threshold_credits');
+        $bundleCredits = $enabled === 1 ? self::creditBundle($data['bundle_credits'] ?? 0, 'bundle_credits') : 0;
+        $monthlyMaxCents = self::boundedMoney($data['monthly_max_cents'] ?? 0, 'monthly_max_cents');
+
+        if ($enabled === 1 && ($thresholdCredits < 1 || $bundleCredits < 1 || $monthlyMaxCents < 1)) {
+            throw new \InvalidArgumentException('Enabled auto top-up requires a threshold, bundle size, and monthly maximum.');
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '125', column '23'). (undefined)

(MissingImport)